sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -44,8 +44,8 @@ abstract class RDG extends UnaryExpression with ExpectsInputTypes with Stateful
   }
 
   @transient protected lazy val seed: Long = child match {
-    case Literal(s, IntegerType) => s.asInstanceOf[Int]
-    case Literal(s, LongType) => s.asInstanceOf[Long]
+    case e if e.dataType == IntegerType => e.eval().asInstanceOf[Int]
+    case e if e.dataType == LongType => e.eval().asInstanceOf[Long]
     case _ => throw new AnalysisException(
       s"Input argument to $prettyName must be an integer, long or null literal.")
```

Comment:
This should be updated accordingly because we are accepting a general expression instead of `literal`.

Reply:
BTW, do you know that why `master` branch doesn't have this `AnalysisException`?

Reply:
The analysis check is done somewhere in the master (https://github.com/apache/spark/pull/30923/files#diff-583171e935b2dc349378063a5841c5b98b30a2d57ac3743a9eccfe7bffcb8f2aR219-R221) but It should throw at least for Scala 2.13 error (https://github.com/apache/spark/pull/30923#discussion_r549707053).

Reply:
I updated the comment, too.